gwy/test.sh

```diff
@@ -11,8 +11,65 @@ fi
 CONTAINER=$1
 HEADERS_FILE="./.headers"
 SOME_RESP_HEADER="SOME-RESPONSE-HEADER"
+
 # Test building the gateway.
-docker run --rm -v=`pwd`:/defs $CONTAINER -f test/test.proto -s Message
+docker run --rm -v=`pwd`:/defs $CONTAINER -f test/test.proto -i . -s Message
+
+# And make sure that we can build the test gateway too.
+docker build -t $CONTAINER-test-gateway gen/grpc-gateway/
+
+# Now run the test container with a prefix in the background
+docker run -p=8080:80 -e 'MESSAGE_PROXY_API-PREFIX=/api/' -e 'MESSAGE_RESPONSE-HEADERS_'${SOME_RESP_HEADER}'=some-value' $CONTAINER-test-gateway &
+
+# Give it a few to start accepting requests
+sleep 5
+
+# Now use curl to make sure we get an expected status.
+# From https://superuser.com/a/442395
+status=`curl -i -s -o $HEADERS_FILE -w "%{http_code}" localhost:8080/api/messages`
```

Comment:
Generally, `$(...)` is preferred over the backtick syntax, as the former can be nested.

Reply:
@jcburley thank you for the code review. I did focus on testing the new functionally rather than improving the test style and readability. The test already had backticks before my changes and I just cloned the existing test code.
Actually, I was expecting having a code review related to duplicated code at gwy/test.sh. I will try to refactor it and push another pull request.

Reply:
Sounds reasonable to me!